<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Trusted Transmission Tool: MainException Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript">
$(document).ready(initResizable);
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body onload='searchBox.OnSelectItem(0);'>
<!-- Generated by Doxygen 1.7.4 -->
<script type="text/javascript"><!--
var searchBox = new SearchBox("searchBox", "search",false,'Search');
--></script>
<div id="top">
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="winIcon.png"/></td>
  <td style="padding-left: 0.5em;">
   <div id="projectname">Trusted Transmission Tool</div>
   <div id="projectbrief">T3</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li id="searchli">
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div>
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
  initNavTree('class_main_exception.html','');
</script>
<div id="doc-content">
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pro-attribs">Protected Attributes</a>  </div>
  <div class="headertitle">
<div class="title">MainException Class Reference</div>  </div>
</div>
<div class="contents">
<!-- doxytag: class="MainException" -->
<p>Throw exceptions and write the errors in Logfile by using <a class="el" href="class_error_logger.html" title="Write errors and messages to Logfile.">ErrorLogger</a>.  
 <a href="class_main_exception.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="mainexception_8h_source.html">mainexception.h</a>&gt;</code></p>
<div class="dynheader">
Inheritance diagram for MainException:</div>
<div class="dyncontent">
<div class="center"><img src="class_main_exception__inherit__graph.png" border="0" usemap="#_main_exception_inherit__map" alt="Inheritance graph"/></div>
<map name="_main_exception_inherit__map" id="_main_exception_inherit__map">
<area shape="rect" id="node3" href="class_user_exception.html" title="Is a Exception class was inherits from MainException and is use to show User errors." alt="" coords="5,83,112,112"/></map>
<center><span class="legend">[<a target="top" href="graph_legend.html">legend</a>]</span></center></div>

<p><a href="class_main_exception-members.html">List of all members.</a></p>
<table class="memberdecls">
<tr><td colspan="2"><h2><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_main_exception.html#a6840c2f8191d9a476d297af6e55d73d5">MainException</a> (const QString errorMsg, const QString errorLocation, int errorType=3)</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Constructor of this exception.  <a href="#a6840c2f8191d9a476d297af6e55d73d5"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_main_exception.html#a433bd3319d974b9bc8a59b31e821414b">~MainException</a> ()  throw ()</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Destructor.  <a href="#a433bd3319d974b9bc8a59b31e821414b"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">virtual const char *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_main_exception.html#aa42afc047fae05e4621b347b861e7f83">what</a> () const   throw ()</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">This method was called from the std::exception to get message.  <a href="#aa42afc047fae05e4621b347b861e7f83"></a><br/></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">QString&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_main_exception.html#acc59bc8ebc1a6a4a73ee9436fd3cdcd6">getMessage</a> ()</td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">Returnss the message of this Exception.  <a href="#acc59bc8ebc1a6a4a73ee9436fd3cdcd6"></a><br/></td></tr>
<tr><td colspan="2"><h2><a name="pro-attribs"></a>
Protected Attributes</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">QString&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_main_exception.html#a43b35dca8a28c7c2bf75ee9300cebbf3">m_errorMessage</a></td></tr>
<tr><td class="mdescLeft">&#160;</td><td class="mdescRight">contain the error message.  <a href="#a43b35dca8a28c7c2bf75ee9300cebbf3"></a><br/></td></tr>
</table>
<hr/><a name="details" id="details"></a><h2>Detailed Description</h2>
<div class="textblock"><p>Throw exceptions and write the errors in Logfile by using <a class="el" href="class_error_logger.html" title="Write errors and messages to Logfile.">ErrorLogger</a>. </p>
<p>Implemented <a class="el" href="class_main_exception.html#aa42afc047fae05e4621b347b861e7f83" title="This method was called from the std::exception to get message.">what()</a> it will called by the standard event handler for error stacktrace. It give the developer or internal error back.</p>
<dl class="author"><dt><b>Author:</b></dt><dd>Otto Ritter <a href="mailto:me@otto-ritter.de">me@otto-ritter.de</a> </dd></dl>
<dl class="version"><dt><b>Version:</b></dt><dd>0.3 </dd></dl>
<dl class="date"><dt><b>Date:</b></dt><dd>04.09.2011 </dd></dl>

<p>Definition at line <a class="el" href="mainexception_8h_source.html#l00038">38</a> of file <a class="el" href="mainexception_8h_source.html">mainexception.h</a>.</p>
</div><hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a6840c2f8191d9a476d297af6e55d73d5"></a><!-- doxytag: member="MainException::MainException" ref="a6840c2f8191d9a476d297af6e55d73d5" args="(const QString errorMsg, const QString errorLocation, int errorType=3)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">MainException::MainException </td>
          <td>(</td>
          <td class="paramtype">const QString&#160;</td>
          <td class="paramname"><em>errorMsg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const QString&#160;</td>
          <td class="paramname"><em>errorLocation</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>errorType</em> = <code>3</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Constructor of this exception. </p>
<p>Call <a class="el" href="class_error_logger.html" title="Write errors and messages to Logfile.">ErrorLogger</a> and write the error in the Logfile. Set Error for called by exception handler virtual method <a class="el" href="class_main_exception.html#aa42afc047fae05e4621b347b861e7f83" title="This method was called from the std::exception to get message.">what()</a>.</p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table class="params">
    <tr><td class="paramname">errorMsg</td><td>QString sets the error message. </td></tr>
    <tr><td class="paramname">errorLocation</td><td>QString sets the location the best is use ErrorPosition::getErrorPosition(ErrorPOS). </td></tr>
    <tr><td class="paramname">errorType</td><td>int the type of the Exception is need for <a class="el" href="class_error_logger.html" title="Write errors and messages to Logfile.">ErrorLogger</a>. </td></tr>
  </table>
  </dd>
</dl>
<dl class="see"><dt><b>See also:</b></dt><dd><a class="el" href="class_error_logger.html#ae3cfe61388f42874c0ed1c87516831cf" title="Sets the message, location, type and write in Logfile.">ErrorLogger::logError()</a> write messages in a logfile. </dd>
<dd>
<a class="el" href="class_error_position.html#a24079597b20ec8abf9df1bbd6ea70be2" title="This static method makes the given position printable.">ErrorPosition::getErrorPosition()</a> to get the possition of the error. </dd></dl>

<p>Definition at line <a class="el" href="mainexception_8cpp_source.html#l00004">4</a> of file <a class="el" href="mainexception_8cpp_source.html">mainexception.cpp</a>.</p>

<p><div class="dynheader">
Here is the call graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="class_main_exception_a6840c2f8191d9a476d297af6e55d73d5_cgraph.png" border="0" usemap="#class_main_exception_a6840c2f8191d9a476d297af6e55d73d5_cgraph" alt=""/></div>
<map name="class_main_exception_a6840c2f8191d9a476d297af6e55d73d5_cgraph" id="class_main_exception_a6840c2f8191d9a476d297af6e55d73d5_cgraph">
<area shape="rect" id="node3" href="class_error_logger.html#a0dc5cc0a4278c1842851e0be41512dd6" title="Returns the instance of this singleton." alt="" coords="256,5,416,35"/><area shape="rect" id="node5" href="class_error_logger.html#ae3cfe61388f42874c0ed1c87516831cf" title="Sets the message, location, type and write in Logfile." alt="" coords="265,59,407,88"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a433bd3319d974b9bc8a59b31e821414b"></a><!-- doxytag: member="MainException::~MainException" ref="a433bd3319d974b9bc8a59b31e821414b" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">MainException::~MainException </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td>  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Destructor. </p>

<p>Definition at line <a class="el" href="mainexception_8cpp_source.html#l00010">10</a> of file <a class="el" href="mainexception_8cpp_source.html">mainexception.cpp</a>.</p>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="acc59bc8ebc1a6a4a73ee9436fd3cdcd6"></a><!-- doxytag: member="MainException::getMessage" ref="acc59bc8ebc1a6a4a73ee9436fd3cdcd6" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">QString MainException::getMessage </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>Returnss the message of this Exception. </p>
<dl class="return"><dt><b>Returns:</b></dt><dd>QString contains the error message. </dd></dl>

<p>Reimplemented in <a class="el" href="class_user_exception.html#a490bbaf258f3502469d59234454db91f">UserException</a>.</p>

<p>Definition at line <a class="el" href="mainexception_8cpp_source.html#l00020">20</a> of file <a class="el" href="mainexception_8cpp_source.html">mainexception.cpp</a>.</p>

<p><div class="dynheader">
Here is the caller graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="class_main_exception_acc59bc8ebc1a6a4a73ee9436fd3cdcd6_icgraph.png" border="0" usemap="#class_main_exception_acc59bc8ebc1a6a4a73ee9436fd3cdcd6_icgraph" alt=""/></div>
<map name="class_main_exception_acc59bc8ebc1a6a4a73ee9436fd3cdcd6_icgraph" id="class_main_exception_acc59bc8ebc1a6a4a73ee9436fd3cdcd6_icgraph">
<area shape="rect" id="node3" href="class_controll_server.html#adb208060bfd8ea47826e588a7870d2aa" title="Decrypt the file that was selected by the user." alt="" coords="273,165,449,195"/><area shape="rect" id="node9" href="class_controll_server.html#a0db50abb897bb5e4b433471273adfde9" title="Encrypt the given file with selected encryption algorithm and the public key from friend..." alt="" coords="275,5,448,35"/><area shape="rect" id="node13" href="class_r_s_a_key_pair.html#a1db01af49d7450c89129cc4296725189" title="This will reset the existing keypair." alt="" coords="295,112,428,141"/><area shape="rect" id="node15" href="class_controll_server.html#a2f39979eddc0c921b2702b331a89a46b" title="Setter for filepath." alt="" coords="240,59,483,88"/><area shape="rect" id="node5" href="class_main_window.html#a1ff65146e7ef9ade8b5f5400e5a7d3d8" title="Starts the decryption by calling doDeCrypt from ControllServer." alt="" coords="533,112,824,141"/><area shape="rect" id="node7" href="class_main_window.html#a8b244be8b7b7db1b08de2a2acb9409db" title="Constructor for the Gui." alt="" coords="876,85,1055,115"/><area shape="rect" id="node11" href="class_main_window.html#a7b847e345bd9ae31311a4a3b55993ada" title="Starts the encryption by calling doEnCrypt from ControllServer." alt="" coords="533,5,824,35"/><area shape="rect" id="node19" href="class_main_window.html#a9294223cb1cec1959e2af702b2dfc2a4" title="Called when the user selects a file for encryption or decryption." alt="" coords="531,59,827,88"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa42afc047fae05e4621b347b861e7f83"></a><!-- doxytag: member="MainException::what" ref="aa42afc047fae05e4621b347b861e7f83" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char * MainException::what </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const  throw ()<code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>This method was called from the std::exception to get message. </p>
<dl class="return"><dt><b>Returns:</b></dt><dd>const char * </dd></dl>

<p>Definition at line <a class="el" href="mainexception_8cpp_source.html#l00015">15</a> of file <a class="el" href="mainexception_8cpp_source.html">mainexception.cpp</a>.</p>

<p><div class="dynheader">
Here is the caller graph for this function:</div>
<div class="dyncontent">
<div class="center"><img src="class_main_exception_aa42afc047fae05e4621b347b861e7f83_icgraph.png" border="0" usemap="#class_main_exception_aa42afc047fae05e4621b347b861e7f83_icgraph" alt=""/></div>
<map name="class_main_exception_aa42afc047fae05e4621b347b861e7f83_icgraph" id="class_main_exception_aa42afc047fae05e4621b347b861e7f83_icgraph">
<area shape="rect" id="node3" href="class_controll_server.html#abea7306fb7e53b040cb0784fa7b3dd60" title="Create the keyfile for the encrypted file." alt="" coords="199,5,383,35"/><area shape="rect" id="node5" href="class_controll_server.html#a0db50abb897bb5e4b433471273adfde9" title="Encrypt the given file with selected encryption algorithm and the public key from friend..." alt="" coords="433,5,607,35"/><area shape="rect" id="node7" href="class_main_window.html#a7b847e345bd9ae31311a4a3b55993ada" title="Starts the encryption by calling doEnCrypt from ControllServer." alt="" coords="656,5,947,35"/></map>
</div>
</p>

</div>
</div>
<hr/><h2>Member Data Documentation</h2>
<a class="anchor" id="a43b35dca8a28c7c2bf75ee9300cebbf3"></a><!-- doxytag: member="MainException::m_errorMessage" ref="a43b35dca8a28c7c2bf75ee9300cebbf3" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">QString <a class="el" href="class_main_exception.html#a43b35dca8a28c7c2bf75ee9300cebbf3">MainException::m_errorMessage</a><code> [protected]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>contain the error message. </p>

<p>Definition at line <a class="el" href="mainexception_8h_source.html#l00077">77</a> of file <a class="el" href="mainexception_8h_source.html">mainexception.h</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following files:<ul>
<li>Source/Helper/<a class="el" href="mainexception_8h_source.html">mainexception.h</a></li>
<li>Source/Helper/<a class="el" href="mainexception_8cpp_source.html">mainexception.cpp</a></li>
</ul>
</div>
</div>
  <div id="nav-path" class="navpath">
    <ul>
      <li class="navelem"><a class="el" href="class_main_exception.html">MainException</a>      </li>
      <li class="footer">Generated on Sat Feb 11 2012 13:11:20 for Trusted Transmission Tool by&#160;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.4 </li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Defines</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>


</body>
</html>
